modules/admaticBidAdapter.js

```diff
@@ -0,0 +1,147 @@
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function(bid) {
+    let isValid = false;
+    if (typeof bid.params !== 'undefined') {
+      let isValidNetworkId = _validateId(getValue(bid.params, 'networkId'));
+      isValid = isValidNetworkId;// && isValidTypeId;
+    }
+
+    if (!isValid) {
+      logError('AdMatic networkId parameters are required. Bid aborted.');
+    }
+    return isValid;
+  },
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const bids = validBidRequests.map(buildRequestObject);
+    const networkId = getValue(validBidRequests[0].params, 'networkId');
+    const currency = getValue(validBidRequests[0].params, 'currency') || 'TRY';
+
+    setTimeout(() => {
+      loadExternalScript(SYNC_URL, BIDDER_CODE);
+    }, bidderRequest.timeout);
+
+    const payload = {
+      'user': {
+        'ua': navigator.userAgent
+      },
+      'blacklist': [],
+      'site': {
+        'page': location.href,
+        'ref': location.origin,
+        'publisher': {
+          'name': location.hostname,
+          'publisherId': networkId
+        }
+      },
+      imp: bids,
+      ext: {
+        'cur': currency,
+        'type': 'admatic'
+      }
+    };
+
+    const payloadString = JSON.stringify(payload);
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL,
+      data: payloadString,
+      options: {
+        contentType: 'application/json'
+      }
+    };
+  },
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: (response, request) => {
+    const body = response.body || response;
+    const bidResponses = [];
+    if (body.data.length > 0) {
+      body.data.forEach(function (bid) {
+        const resbid = {
+          requestId: bid.id,
+          cpm: bid.price,
+          width: bid.width,
+          height: bid.height,
+          currency: body.cur,
+          netRevenue: true,
+          ad: bid.party_tag,
+          creativeId: bid.creative_id,
+          meta: {
+            advertiserDomains: bid && bid.adomain ? bid.adomain : []
+          },
+          ttl: 360,
+          bidder: 'admatic',
+          timeToRespond: 1,
+          requestTimestamp: 1
+        };
+        bidResponses.push(resbid);
+      });
+    };
+    return bidResponses;
+  }
+};
+
+function buildRequestObject(bid) {
+  const reqObj = {};
+  reqObj.size = getSizes(bid);
+  reqObj.id = getBidIdParameter('bidId', bid);
+  reqObj.floor = getValue(bid.params, 'floor') || 0.01;
```

Comment:
I guess this will be flagged as an integration with the prebid.js floor module is mandatory if floors should be supported

Reply:
Okay. I solved this issue.